libenv/sysinfo.c

```diff
@@ -1085,14 +1085,24 @@ static void OSClasses(EvalContext *ctx)
 
     if (os_release_json != NULL)
     {
-        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "os_release", os_release_json, CF_DATA_TYPE_CONTAINER, "source=agent,os,release,derived_from=/etc/os-release");
+        EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, "os_release",
+                                      os_release_json, CF_DATA_TYPE_CONTAINER,
+                                      "source=agent,os,release,derived_from=/etc/os-release");
         const char *os_release_id = JsonObjectGetAsString(os_release_json, "ID");
-        if(os_release_id != NULL)
+        const char *os_release_version = JsonObjectGetAsString(os_release_json, "VERSION");
+        char os_release_flavor[256];
+        snprintf(os_release_flavor, 256, "%s_%s", os_release_id, os_release_version);
+        char *dot_pointer = os_release_flavor;
+        while( (dot_pointer = strchr(dot_pointer, '.')) != NULL )
         {
-            if(strcmp(os_release_id, "coreos") == 0)
-            {
-                EvalContextClassPutHard(ctx, "coreos", "inventory,attribute_name=none,source=agent");
-            }
+            *dot_pointer = '_';
```

Comment:
I thought you were setting classes. I believe it applies, since these variable values are canonified class-like. Don't ask me why.

Plus you'll probably have to set classes as well for full CoreOS support. Here are the classes on Centos 7:
```
centos                                                       inventory,attribute_name=none,source=agent,hardclass
centos_7                                                     inventory,attribute_name=none,source=agent,hardclass
centos_7_3                                                   inventory,attribute_name=none,source=agent,hardclass
```